game-core/src/main/java/games/strategy/engine/framework/ui/MainFrame.java

```diff
@@ -0,0 +1,65 @@
+package games.strategy.engine.framework.ui;
+
+import com.google.common.base.Preconditions;
+import games.strategy.engine.framework.GameRunner;
+import games.strategy.engine.framework.lookandfeel.LookAndFeelSwingFrameListener;
+import games.strategy.engine.framework.startup.mc.SetupPanelModel;
+import games.strategy.engine.framework.startup.ui.panels.main.MainPanelBuilder;
+import games.strategy.engine.framework.startup.ui.panels.main.game.selector.GameSelectorModel;
+import games.strategy.engine.framework.ui.background.BackgroundTaskRunner;
+import java.awt.event.WindowEvent;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.WindowConstants;
+import org.triplea.swing.JFrameBuilder;
+
+/** Represents the outermost JFrame, maintains the reference to it and controls access. */
+public class MainFrame {
+
+  private static MainFrame instance;
+
+  private JFrame mainFrame;
+
+  private MainFrame(
+      final SetupPanelModel setupPanelModel, final GameSelectorModel gameSelectorModel) {
+    mainFrame =
+        JFrameBuilder.builder()
+            .title("TripleA")
+            .windowClosedAction(GameRunner::exitGameIfFinished)
+            .build();
+    BackgroundTaskRunner.setMainFrame(mainFrame);
+
+    LookAndFeelSwingFrameListener.register(mainFrame);
```

Comment:
I think those static actions should be moved up one call layer:
I'f this constructor is ever called tice this could result in weird problems

Reply:
The constructor is private, the factory method checks that the static instance is not yet set. One level higher is 'GameRunner', part of the goal is to remove anything UI or swing related out of GameRunner.

Reply:
By "one level higher" I was actually reffering to the static method inside this class rather than the constructor